pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -464,6 +468,26 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	// needed for luksOpen synchronization when devices are encrypted and the osd is prepared with LVM
 	hostIPC := osdProps.storeConfig.EncryptedDevice || osdProps.encrypted
 
+	osdLabels := c.getOSDLabels(osd, failureDomainValue, osdProps.portable)
+
+	if osd.ExportService {
+		osdService, err := c.createOSDService(osd, osdLabels)
+		if err != nil {
+			logger.Errorf("failed to configure osd service for osd.%d. %v", osd.ID, err)
```

Comment:
Need to return an error here? We won't be able to export the service in this error case.

Reply:
ah. Right. I changed this in one of the previous commits. Looks like the old code creeped in somehow!